File: Miscellaneous/Python/268. Missing Number.py
class Solution:
    def missingNum(self, arr):
        sum, n = 0, len(arr)
        sum = (n * (n+1)) // 2

        for num in arr:
            sum -= num
        
        return sum

obj = Solution()
arr = [9,6,4,2,3,5,7,0,1]
ans = obj.missingNum(arr)
print(ans)

File: Miscellaneous/Golang/268. Missing Number.go
package main

import "fmt"

func main() {

	arr := []int{9, 6, 4, 2, 3, 5, 7, 0, 1}
	result := missingNumber(arr)
	fmt.Println(result)
}

func missingNumber(arr []int) int {
	n := len(arr)
	sum := n * (n + 1) / 2

	for _, num := range arr {
		sum -= num
	}
	return sum
}
